haystack_experimental/components/builders/answer_builder.py

```diff
@@ -0,0 +1,130 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Any, Dict, List, Optional, Union
+
+from haystack import Document, component, logging
+from haystack.components.builders.answer_builder import AnswerBuilder as HaystackAnswerBuilder
+from haystack.dataclasses.chat_message import ChatMessage
+
+from haystack_experimental.dataclasses import GeneratedAnswer
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class AnswerBuilder(HaystackAnswerBuilder):
+    """
+    Converts a query and Generator replies into a `GeneratedAnswer` object.
+
+    AnswerBuilder parses Generator replies using custom regular expressions.
+    Check out the usage example below to see how it works.
+    Optionally, it can also take documents and metadata from the Generator to add to the `GeneratedAnswer` object.
+    AnswerBuilder works with both non-chat and chat Generators.
+
+    ### Usage example
+
+    ```python
+    from haystack.components.builders import AnswerBuilder
+
+    builder = AnswerBuilder(pattern="Answer: (.*)")
+    builder.run(query="What's the answer?", replies=["This is an argument. Answer: This is the answer."])
+    ```
+    """
+
+    @component.output_types(answers=List[GeneratedAnswer])
+    def run(  # pylint: disable=too-many-positional-arguments
```

Comment:
Just to double check we still have this method here just to update this line 
`@component.output_types(answers=List[GeneratedAnswer])` to use the new `GeneratedAnswer` from haystack experimental right?